File: frontend/lib/api/earnings.ts
import { get } from "./client"
import type { EarningsCalendarResponse } from "@/types/earnings"

export async function getStockEarnings(
  symbol: string
): Promise<EarningsCalendarResponse> {
  return get<EarningsCalendarResponse>(
    `/api/v1/stocks/${symbol.toUpperCase()}/earnings`
  )
}

export async function getEarningsCalendar(
  symbols: string[]
): Promise<EarningsCalendarResponse> {
  if (symbols.length === 0) return { items: [] }
  return get<EarningsCalendarResponse>(
    `/api/v1/earnings/calendar?symbols=${symbols.join(",")}`
  )
}

File: backend/app/api/v1/endpoints/earnings.py
from fastapi import APIRouter, Query

from app.schemas.earnings import EarningsCalendarResponse
from app.services.earnings_service import (
    get_earnings_for_symbols,
)

router = APIRouter()


@router.get(
    "/calendar",
    response_model=EarningsCalendarResponse,
    summary="Get upcoming earnings for multiple symbols",
)
async def get_earnings_calendar(
    symbols: str = Query(
        ..., description="Comma-separated symbols, e.g. AAPL,MSFT,NVDA"
    ),
) -> EarningsCalendarResponse:
    """Fetch upcoming earnings for portfolio/watchlist symbols."""
    sym_list = [s.strip().upper() for s in symbols.split(",") if s.strip()][:50]
    events = await get_earnings_for_symbols(sym_list)
    return EarningsCalendarResponse(items=events)
